### ISSUE TITLE
CMake include messes with compiler options

### ISSUE THREAD

**Issue by nschloe** [CGAL/cgal#2849]
Here's a bug in CGAL's cmake system that took hours to pin down, but hopefully the following MWE will make things clear and help with a fix:
```cmake
cmake_minimum_required(VERSION 3.0)

project(foobar CXX)

FIND_PACKAGE(CGAL REQUIRED)
include(${CGAL_USE_FILE})

include(CheckCXXCompilerFlag)
CHECK_CXX_COMPILER_FLAG("-std=c++14" HAS_CXX14)
message("has_cxx14? <${HAS_CXX14}>")
```
When configuring with `CXX=g++`, this correctly gives
```
has_cxx14? <1>
```
However, when configuring with `CXX=clang++`, it _incorrectly_ gives
```
has_cxx14? <>
```
When putting the compiler flag check _before_  `include(${CGAL_USE_FILE})`, it also works fine. This has be suspect that `${CGAL_USE_FILE}` tries to mess around with the compiler.

**Comment by sloriot** [comment]
I can reproduce the issue with the 4.11 release but not with master. I believe this is also fixed by #1436

**Comment by nschloe** [comment]
Yes, this PR looks quite massive. I'll keep my eyes open for 4.12 then. Thanks again!
